### PR TITLE
Fix vulnerability detector content database generation

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#25974|

## Description

- Fixing checkout to target 4.10.0 branch step. 
- Workaround for disparity between 4.8.0 and 5.0.0 github workflows

### CODE REVIEW

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 1]
```diff
@@ -62,6 +64,16 @@ jobs:
...
 
+          cp -r /tmp/{actions,workflows} .github/
+
+          template=src/wazuh_modules/vulnerability_scanner/indexer/template/index-template.json
+          if [ -f "$template" ]; then
+            if [ ! -d "templates" ]; then
+              mkdir templates
+            fi
+            cp "$template" templates/vd_states_template.json
+          fi
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:75]: This should be done in another step

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 2]
```diff
@@ -48,11 +48,13 @@ jobs:
       # Checkout to the tag. If it doesn't exist, continue with the branch
       - name: Checkout to ${{ matrix.content_version }}
         run: |
+          cp -r .github/{actions,workflows} /tmp
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:51]: Add a comment explaining this

File: .github/workflows/vulnerability-scanner-generate-database.yml (hunk 1 above)
Comment (sebasfalcone) [reply .github/workflows/vulnerability-scanner-generate-database.yml:75]: Add a comment explaining this

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 3]
```diff
@@ -62,6 +64,16 @@ jobs:
           fi
           echo "Git branch: $(git branch | grep "*")"
 
+          cp -r /tmp/{actions,workflows} .github/
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:67]: Add a comment explaining this

File: .github/workflows/vulnerability-scanner-generate-database.yml (hunk 1 above)
Comment (MiguelazoDS) [reply .github/workflows/vulnerability-scanner-generate-database.yml:75]: Added some comments, although is not too much complex to the already existent logic there.

File: .github/workflows/vulnerability-scanner-generate-database.yml [hunk 4]
```diff
@@ -62,6 +65,21 @@ jobs:
           fi
           echo "Git branch: $(git branch | grep "*")"
 
+          # Restoring workflows and actions.
```
Comment (sebasfalcone) [inline .github/workflows/vulnerability-scanner-generate-database.yml:68]:
```suggestion
          # We restore the actions from master 
          # to re-use them independently from the checkout branch
```

**Review by sebasfalcone — APPROVED** [review]
@MiguelazoDS Let's revert the last commit to restore the cronjob

**Review by sebasfalcone — COMMENTED** [review]
[This step](https://github.com/wazuh/wazuh/blob/ea2a21c45b708402a38f5e9da8d00a6d9a1b06a3/.github/workflows/vulnerability-scanner-generate-database.yml#L115-L143) of the worflow remained without use. We could move that logic to the generation step

**Review by sebasfalcone — APPROVED** [review]
LGTM!

**Pull request merged**